### PR TITLE
feat(table): fixed logic for showing/hiding no result message

### PR DESCRIPTION
**Pull request by mJarsater**
**Describe pull-request**  
 -  Fixed logic for showing/hiding no result message
 - Introduces slot for no result message
 - Introduces prop for no result message. 

**Solving issue**  
Fixes: https://tegel.atlassian.net/browse/DTS-2044

**How to test**  
1. Go to Table -> Filtering
2. Search for "jp"
3. Check that the no result message works.

**Suggested test steps**
- [ ] Browser testing (Chrome, Safari, Firefox) 
- [ ] Keyboard operability
- [ ] Interactive elements have labels.
- [ ] Storybook controls
- [ ] Design/controls/props is aligned with other components 
- [ ] Dark/light mode and variants 
- [ ] Input fields – values should be displayed properly 
- [ ] Events

### CODE REVIEW

File: core/src/components/table/table-body/table-body.tsx [hunk 1]
```diff
@@ -75,6 +81,9 @@ export class TdsTableBody {
   /** Prop for showcase of rendering JSON in body-data, just for presentation purposes */
   @Prop() enableDummyData: boolean = false;
 
+  /** Prop for showcase of rendering JSON in body-data, just for presentation purposes */
```
Comment (nathalielindqvist) [inline core/src/components/table/table-body/table-body.tsx:84]:
```suggestion
  /** Prop for no result message when using filtering */
```

File: core/src/components/table/table-body/table-body.tsx [hunk 2]
```diff
@@ -63,6 +64,11 @@ const relevantTableProps: InternalTdsTablePropChange['changed'] = [
   'enableExpandableRows',
 ];
 
+
+/**
+ * @slot default - Slot for <tds-table-body-row>
```
Comment (mJarsater) [inline core/src/components/table/table-body/table-body.tsx:69]:
```suggestion
 * @slot default - Default slot for the Table Body, used by <tds-table-body-row>
```

**Review by nathalielindqvist — APPROVED** [review]
LGTM!

File: core/src/components/table/table-body/table-body.tsx [hunk 3]
```diff
@@ -367,13 +377,12 @@ export class TdsTableBody {
             ))}
           </tds-table-body-row>
         ))}
-        {this.showNoResultsMessage && (
-          <tr>
+          <tr hidden={!this.showNoResultsMessage}>
             <td class="tds-table__info-message" colSpan={this.columnsNumber}>
-              Unfortunately, no data matches your search term &#128533;
+              <slot name="no-result"/>
```
Comment (theJohnnyMe) [inline core/src/components/table/table-body/table-body.tsx:382]: Nice improvement!

**Review by theJohnnyMe — DISMISSED** [review]
🚀

**Review by alexandereneroth — APPROVED** [review]

File: core/src/components/table/table-body/table-body.tsx [hunk 4]
```diff
@@ -63,6 +64,11 @@ const relevantTableProps: InternalTdsTablePropChange['changed'] = [
   'enableExpandableRows',
 ];
 
+
+/**
+ * @slot default - Default slot for the Table Body, used by <tds-table-body-row>
```
Comment (theJohnnyMe) [inline core/src/components/table/table-body/table-body.tsx:69]:
```suggestion
 * @slot <default> - Default slot for the Table Body, used by <tds-table-body-row>
```

**Review by theJohnnyMe — APPROVED** [review]
🤘

**Review by alexandereneroth — APPROVED** [review]

**Pull request merged**